iabtcf-encoder/src/main/java/com/iabtcf/encoder/BitFieldEncoder.java

```diff
@@ -0,0 +1,166 @@
+package com.iabtcf.encoder;
+
+/*-
+ * #%L
+ * IAB TCF Java Encoder Library
+ * %%
+ * Copyright (C) 2020 IAB Technology Laboratory, Inc
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import java.util.BitSet;
+import java.util.Iterator;
+
+import com.iabtcf.FieldDefs;
+import com.iabtcf.utils.IntIterable;
+import com.iabtcf.utils.IntIterator;
+
+/**
+ * Encodes a bit / range field, typically used for encoding vendor lists.
+ */
+class BitFieldEncoder {
+    private final BitSet vendors;
+    private int maxVendorId;
```

Comment:
can we call them `max` and `bitSet`, lets not be specific to vendors

Reply:
`maxVendorId` might be ok, just because this encoder can only be used to encode a vendor  "vector". None of the other vectors have a range encoding.

Reply:
you know, i went ahead and tried to do this and it kinda was less clear. there is so much vendor-specific lingo in this encoder... it can certainly be made generic but it doesn't make sense in context of this library.

Reply:
let me know if you feel strongly about this, i can certainly reconsider.

Reply:
Maybe this class should be renamed to VendorFieldEncoder?